### ISSUE TITLE
Exception has occurred: OutOfRangeError RandomShuffleQueue '_2_shuffle_batch/random_shuffle_queue' is closed and has insufficient elements (requested 10, current size 0)

### ISSUE THREAD

**Issue by Halleyawoo** [carrenD/Medical-Cross-Modality-Domain-Adaptation#16]
There is no problem with my file path, but this error occurred.

Exception has occurred: OutOfRangeError
RandomShuffleQueue '_2_shuffle_batch/random_shuffle_queue' is closed and has insufficient elements (requested 10, current size 0)
	 [[Node: shuffle_batch = QueueDequeueManyV2[component_types=[DT_FLOAT, DT_FLOAT, DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/device:CPU:0"](shuffle_batch/random_shuffle_queue, shuffle_batch/n)]]
	 [[Node: shuffle_batch/_67 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_5_shuffle_batch", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

Caused by op u'shuffle_batch', defined at:
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/hfcui/.vscode-server/extensions/ms-python.python-2020.9.114305/pythonFiles/lib/python/debugpy/__main__.py", line 45, in <module>
    cli.main()
  File "/home/hfcui/.vscode-server/extensions/ms-python.python-2020.9.114305/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/home/hfcui/.vscode-server/extensions/ms-python.python-2020.9.114305/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 267, in run_file
    runpy.run_path(options.target, run_name=compat.force_str("__main__"))
  File "/usr/lib/python2.7/runpy.py", line 252, in run_path
    return _run_module_code(code, init_globals, run_name, path_name)
  File "/usr/lib/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/media/nomachine/project/Medical-Cross-Modality-Domain-Adaptation/train_segmenter.py", line 88, in <module>
    main()
  File "/media/nomachine/project/Medical-Cross-Modality-Domain-Adaptation/train_segmenter.py", line 82, in main
    trainer.train(output_path = output_path, training_iters = training_iters, epochs = epochs, restore = True, restored_path = restored_path)
  File "/media/nomachine/project/Medical-Cross-Modality-Domain-Adaptation/source_segmenter.py", line 466, in train
    feed_all, feed_fid = self.next_batch(self.train_queue)
  File "/media/nomachine/project/Medical-Cross-Modality-Domain-Adaptation/source_segmenter.py", line 351, in next_batch
    num_threads = num_threads, min_after_dequeue = min_after_dequeue)
  File "/home/hfcui/.local/lib/python2.7/site-packages/tensorflow/python/training/input.py", line 1225, in shuffle_batch
    name=name)
  File "/home/hfcui/.local/lib/python2.7/site-packages/tensorflow/python/training/input.py", line 796, in _shuffle_batch
    dequeued = queue.dequeue_many(batch_size, name=name)
  File "/home/hfcui/.local/lib/python2.7/site-packages/tensorflow/python/ops/data_flow_ops.py", line 464, in dequeue_many
    self._queue_ref, n=n, component_types=self._dtypes, name=name)
  File "/home/hfcui/.local/lib/python2.7/site-packages/tensorflow/python/ops/gen_data_flow_ops.py", line 2418, in _queue_dequeue_many_v2
    component_types=component_types, timeout_ms=timeout_ms, name=name)
  File "/home/hfcui/.local/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/hfcui/.local/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 2956, in create_op
    op_def=op_def)
  File "/home/hfcui/.local/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1470, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

OutOfRangeError (see above for traceback): RandomShuffleQueue '_2_shuffle_batch/random_shuffle_queue' is closed and has insufficient elements (requested 10, current size 0)
	 [[Node: shuffle_batch = QueueDequeueManyV2[component_types=[DT_FLOAT, DT_FLOAT, DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/device:CPU:0"](shuffle_batch/random_shuffle_queue, shuffle_batch/n)]]
	 [[Node: shuffle_batch/_67 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_5_shuffle_batch", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
  File "/media/nomachine/project/Medical-Cross-Modality-Domain-Adaptation/source_segmenter.py", line 478, in train
    batch, fid = sess.run([feed_all, feed_fid])
  File "/media/nomachine/project/Medical-Cross-Modality-Domain-Adaptation/train_segmenter.py", line 82, in main
    trainer.train(output_path = output_path, training_iters = training_iters, epochs = epochs, restore = True, restored_path = restored_path)
  File "/media/nomachine/project/Medical-Cross-Modality-Domain-Adaptation/train_segmenter.py", line 88, in <module>
    main()

**Comment by cheng-01037** [comment]
I would suggest you to set a breakpoint in the script and see if these tfrecords files are readable. You might also want to check if it is the issue of relative path vs absolute path.

**Comment by Halleyawoo** [comment]
> I would suggest you to set a breakpoint in the script and see if these tfrecords files are readable. You might also want to check if it is the issue of relative path vs absolute path.

Thx

**Comment by 18801109112** [comment]
hi, I also met the same problem. Have you solved your problem
THX

**Comment by Faunder-fld** [comment]
Here the best answer I had ever found: [https://github.com/carrenD/ummkd/issues/9](https://github.com/carrenD/ummkd/issues/9)
